sway-fmt-v2/src/fmt.rs

```diff
@@ -46,19 +50,23 @@ impl Formatter {
 
         // Formatted code will be pushed here with raw newline stlye.
         // Which means newlines are not converted into system-specific versions by apply_newline_style
-        let mut raw_formatted_code = String::new();
+        // Use the length of src as a hint of the memory size needed for raw_formatted_code, 
+        // which will reduce the number of reallocations
+        let mut raw_formatted_code = String::with_capacity(src_len);
```

Comment:
Use the length of src as a hint of the memory size needed for `raw_formatted_code`, which will reduce the number of reallocations.